cloudbuild/cloudbuild-cdn.yaml

```diff
@@ -79,49 +79,63 @@ steps:
       - '-c'
       - |
+        # backslash-escape any non-delimiter whitespace and all other characters that
+        # have special meaning to the shell, e.g. globs, parenthesis, ampersands, etc.
+        INSTANCES='${_GR4VY_INSTANCES}'
+
         if [ "${BRANCH_NAME}" = "main" ]; then
-          echo "Triggering deploy of main branch to spider-sandbox..."
-          curl \
-            -H "Accept: application/vnd.github.v3+json" \
-            -H "Authorization: token $$GITHUB_TOKEN" \
-            -d '{"ref":"${COMMIT_SHA}","environment":"spider-sandbox","auto_merge":false,"required_contexts":[],"payload":{"substitutions":{"region":"us-central1"},"config_path":"cloudbuild/deploy-env.yaml"}}' \
-            https://api.github.com/repos/gr4vy/${REPO_NAME}/deployments
+          for instance in $${INSTANCES}; do
+            echo "Triggering deploy of main branch to $${instance}-sandbox..."
+            docker run --rm -e GITHUB_TOKEN gcr.io/${PROJECT_ID}/deployer:latest --id $${instance} --sandbox --repo ${REPO_NAME} --ref ${COMMIT_SHA} --skip-checks
+          done
         else
-          echo "This is the ${BRANCH_NAME} branch. Not deploying to spider-sandbox."
+          echo "This is the ${BRANCH_NAME} branch. Not deploying to sandbox environments."
         fi;
 
-  - id: 'deploy to spider-production'
-    name: 'curlimages/curl:latest'
-    entrypoint: ash
+  - id: 'deploy to production environments'
+    name: 'docker:20.10.5'
+    entrypoint: 'ash'
     secretEnv:
       - GITHUB_TOKEN
     args:
       - '-c'
       - |
+        # backslash-escape any non-delimiter whitespace and all other characters that
+        # have special meaning to the shell, e.g. globs, parenthesis, ampersands, etc.
+        INSTANCES='${_GR4VY_INSTANCES}'
+
         if [ "${BRANCH_NAME}" = "main" ]; then
-          echo "Triggering deploy of main branch to spider-production..."
-          curl \
-            -H "Accept: application/vnd.github.v3+json" \
-            -H "Authorization: token $$GITHUB_TOKEN" \
-            -d '{"ref":"${COMMIT_SHA}","environment":"spider-production","auto_merge":false,"required_contexts":[],"payload":{"substitutions":{"region":"us-central1"},"config_path":"cloudbuild/deploy-env.yaml"}}' \
-            https://api.github.com/repos/gr4vy/${REPO_NAME}/deployments
+          for instance in $${INSTANCES}; do
+            echo "Triggering deploy of main branch to $${instance}-production..."
+            docker run --rm -e GITHUB_TOKEN gcr.io/${PROJECT_ID}/deployer:latest --id $${instance} --production --repo ${REPO_NAME} --ref ${COMMIT_SHA} --skip-checks
+          done
         else
-          echo "This is the ${BRANCH_NAME} branch. Not deploying to spider-production."
+          echo "This is the ${BRANCH_NAME} branch. Not deploying to production environments."
         fi;
 
 substitutions:
   _NODE_VERSION: '14'
   _ARTIFACT_STORAGE_BUCKET: ''
   _CACHE_BUCKET: ''
   _GITHUB_TOKEN_LOCATION: ''
+  _GR4VY_INSTANCES: ''
```

Comment:
empty instances?

Reply:
Because this repo is public we leave the definition here empty to avoid advertising the instance names.

The value for this (and other substitutions) is set within the Google Cloud Build trigger that starts running this job.

Reply:
would it make sense to store it in the repo's settings like a secret env var?

Reply:
I'm not sure I understand... Do you mean in the GitHub repo's secrets?
This is a Cloud Build job, so doesn't use any secrets from GitHub.

We could add [`availableSecrets`](https://cloud.google.com/build/docs/securing-builds/use-secrets) block to this file to allow the Cloud Build step to collect a value from Google Secret Manager, but the values for the instances would still need to be set in Secret Manager, just as they are in the Cloud Build trigger.

Effectively it would just move the values from Cloud Build to Secret Manager. Given that these values are really secret, we just don't want to make them very public, and that there is a (small) cost for using Secret Manager, I'm not sure it's a better approach to having the values defined in the trigger.

Reply:
> This is a Cloud Build job, so doesn't use any secrets from GitHub.
ignore me, I thought this was a GB action 🙄 

Thanks for the explanation

Reply:
No worries.